include/openthread/tcp.h

```diff
@@ -0,0 +1,616 @@
+ * @param[in]  aEndpoint            A pointer to the TCP endpoint structure whose callback to set.
+ * @param[in]  aBytesAckedCallback  The new callback.
+ *
+ * @retval OT_ERROR_NONE    Successfully set the callback.
+ * @retval OT_ERROR_FAILED  Failed to set the callback.
+ */
+otError otTcpSetBytesAckedCallback(otInstance *aInstance, otTcpEndpoint *aEndpoint, otTcpBytesAcked aBytesAckedCallback);
+
+/**
+ * Sets the "send ready" callback on a TCP endpoint.
+ *
+ * @param[in]  aInstance           A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint           A pointer to the TCP endpoint structure whose callback to set.
+ * @param[in]  aSendReadyCallback  The new callback.
+ *
+ * @retval OT_ERROR_NONE    Successfully set the callback.
+ * @retval OT_ERROR_FAILED  Failed to set the callback.
+ */
+otError otTcpSetSendReadyCallback(otInstance *aInstance, otTcpEndpoint *aEndpoint, otTcpSendReady aSendReadyCallback);
+
+/**
+ * Sets the "receive available" callback on a TCP endpoint.
+ *
+ * @param[in]  aInstance                  A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint                  A pointer to the TCP endpoint structure whose callback to set.
+ * @param[in]  aReceiveAvailableCallback  The new callback.
+ *
+ * @retval OT_ERROR_NONE    Successfully set the callback.
+ * @retval OT_ERROR_FAILED  Failed to set the callback.
+ */
+otError otTcpSetReceiveAvailableCallback(otInstance *aInstance, otTcpEndpoint *aEndpoint, otTcpReceiveAvailable aReceiveAvailableCallback);
+
+/**
+ * Sets the "disconnected" callback on a TCP endpoint.
+ *
+ * @param[in]  aInstance                A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint                A pointer to the TCP endpoint structure whose callback to set.
+ * @param[in]  aDisconnectedCallback    The new callback.
+ *
+ * @retval OT_ERROR_NONE    Successfully set the callback.
+ * @retval OT_ERROR_FAILED  Failed to set the callback.
+ */
+otError otTcpSetDisconnectedCallback(otInstance *aInstance, otTcpEndpoint *aEndpoint, otTcpDisconnected aDisconnectedCallback);
+
+/**
+ * Binds the TCP endpoint to an IP address and port.
+ *
+ * @param[in]  aInstance   A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint   A pointer to the TCP endpoint structure to bind.
+ * @param[in]  aSockName   The address and port to which to bind this TCP endpoint.
+ *
+ * @retval OT_ERROR_NONE    Successfully bound the TCP endpoint.
+ * @retval OT_ERROR_FAILED  Failed to bind the TCP endpoint.
+ */
+otError otTcpBind(otInstance *aInstance, otTcpEndpoint *aEndpoint, const otSockAddr *aSockName);
+
+/**
+ * This enumeration defines flags passed to @p otTcpConnect.
+ */
+enum {
+    OT_TCP_CONNECT_NO_FAST_OPEN = 1 << 0,
+};
+
+/**
+ * Records the remote host and port for this connection. By default TCP Fast
```

Comment:
nit: have one-line brief in own block

Reply:
Done. I'm not 100% what you meant by "in own block", can you please check that what I've done is acceptable?